solr/core/src/java/org/apache/solr/core/SchemaCodecFactory.java

```diff
@@ -145,4 +151,34 @@ public Codec getCodec() {
     assert core != null : "inform must be called first";
     return codec;
   }
+
+  /**
+   * This class exists because Lucene95HnswVectorsFormat's getMaxDimensions method is final and we
+   * need to workaround that constraint to allow more than the default number of dimensions
+   */
```

Comment:
3/3 ... this is a nice workaround to get our max dimensions.

Perhaps overengineering but something like https://github.com/apache/lucene/blob/releases/lucene/9.8.0/lucene/core/src/test/org/apache/lucene/index/TestFilterCodecReader.java#L30 could be added as a test to ensure all that needs delegating is delegated. 

And perhaps the workaround class could be final itself and even private?

Though that is both subjective of course.

Reply:
> Perhaps overengineering but something like https://github.com/apache/lucene/blob/releases/lucene/9.8.0/lucene/core/src/test/org/apache/lucene/index/TestFilterCodecReader.java#L30 could be added as a test to ensure all that needs delegating is delegated.

it does feel like overkill for this

> And perhaps the workaround class could be final itself and even private?

sure, this makes sense. I was concerned with the name being involved in the SPI lookup somehow making things weird. I can test it out